src/shares/PaymentHub.sol

```diff
@@ -113,8 +173,31 @@ contract PaymentHub {
     }
 
     function payFromEtherAndNotify(address recipient, uint256 xchfamount, bytes calldata ref) payable external {
-        payFromEther(recipient, xchfamount);
-        ITokenReceiver(recipient).onTokenTransfer(address(currency), msg.sender, xchfamount, ref);
+        // Check if the brokerbot has setting to keep ETH
+        if (hasSettingKeepEther(recipient)) {
+            uint256 priceInEther = getPriceInEther(xchfamount, recipient);
```

Comment:
Maybe we could use here directly `return getPriceInEtherFromOracle(amountOfXCHF);` , because that's what getPriceInEther returns when hasSettingKeepEther is true, but also it checks hasSettingKeepEther again and could add to the cost.